src/ShellSdk.ts

```diff
@@ -125,6 +126,35 @@ export class ShellSdk {
     );
   }
 
+  public setIgnoredOrigins(ignoredOrigins: string[] = []) {
+    this.ignoredOrigins = ignoredOrigins;
+  }
+
+  public addIgnoredOrigin(url: string) {
+    let urlObj: URL;
+    try {
+      urlObj = new URL(url);
+    } catch {
+      return;
+    }
+    this.ignoredOrigins.push(urlObj.origin);
+  }
+
+  public removeIgnoredOrigin(url: string) {
+    let urlObj: URL;
+    try {
+      urlObj = new URL(url);
+    } catch {
+      return;
+    }
+    const idxToRemove = this.ignoredOrigins.findIndex(
```

Comment:
Why not to use `filter` right here ?

Reply:
Code is using same implementation as the `removeAllowedOrigin` method. Out of consistency will keep as it is for now and plan a refactoring :), but agree could be simplified thanks for the input 👍